### PR TITLE
Component Proposal: Progress Bar

### PR DESCRIPTION
**Pull request by service-paradis**
## Proposed Changes

- Add a new compenent for progress
- Wrapper around Bulma progress (which is Bulma Style over native progress element)

captured

### Works with types
captured

### Works with sizes
captured (1)

### Value customization
captured (2)

### Label can be anything using slot
captured (1)

Ref: #953, #954

### CODE REVIEW

**Comment by jtommy** [comment]
Looks nice! I'll review it soon

File: src/components/progress/Progress.vue [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                const val = this.toFixed(this.value * this.max / 100)
+                return `${val}%`
+            }
+            const val = this.toFixed(this.value)
+            return val
+        }
+    },
+    watch: {
+        value(value) {
+            this.setValue(value)
```
Comment (jtommy) [inline src/components/progress/Progress.vue]: In order to remove the `setValue` call in mounted hook, you might add `immediate` option in the watcher, for example:
```
value: {
    immediate: true,
    handler(value) {
      ...
    }
  }
```

File: src/components/progress/Progress.vue [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        }
+    },
+    methods: {
+        /**
+        * When value is changed back to undefined, value of native progress get reset to 0.
+        * Need to add and remove the value attribute to have the indeterminate or not.
+        */
+        setValue(value) {
+            if (this.isIndeterminate) {
+                this.$refs.progress.removeAttribute('value')
```
Comment (jtommy) [inline src/components/progress/Progress.vue]: Why do you need to work using refs ?

File: src/components/progress/Progress.vue [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+export default {
+    name: 'BProgress',
+    props: {
+        type: {
+            type: String | Object,
+            default: 'is-darkgrey'
+        },
+        size: String,
+        value: {
+            type: undefined | Number,
```
Comment (jtommy) [inline src/components/progress/Progress.vue]: There is no need to specify undefined type

File: src/components/progress/Progress.vue [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    </div>
+</template>
+
+<script>
+
+export default {
+    name: 'BProgress',
+    props: {
+        type: {
+            type: String | Object,
```
Comment (jtommy) [inline src/components/progress/Progress.vue]: Is a really correct syntax ? I usually use [String, Object] array syntax.

File: src/components/progress/Progress.vue (hunk 2 above)
Comment (service-paradis) [reply src/components/progress/Progress.vue]: In order to have an indeterminate progress bar, the `value` attribute need to be removed completely fro the component. If we set the value to `null` or something, the value is considered to be 0 and wont be displayed as indeterminate.

Is there another way to remove an attribute without $refs?

File: src/components/progress/Progress.vue (hunk 3 above)
Comment (service-paradis) [reply src/components/progress/Progress.vue]: My mistake! i'll fix it!

File: src/components/progress/Progress.vue (hunk 4 above)
Comment (service-paradis) [reply src/components/progress/Progress.vue]: Your syntax is more common for Vue props. I'll fix it!

File: src/components/progress/Progress.vue (hunk 1 above)
Comment (service-paradis) [reply src/components/progress/Progress.vue]: The problem by using the `immediate` property is the function will be called before vue's mount hook. That means that $refs won't be visible from the watch function.
You can see my next comment about the usage of $refs.

File: src/components/progress/Progress.vue (hunk 2 above)
Comment (jtommy) [reply src/components/progress/Progress.vue]: No, it isn't. I think it's ok at the moment

File: src/components/progress/Progress.vue (hunk 1 above)
Comment (jtommy) [reply src/components/progress/Progress.vue]: You're right!

**Pull request merged**